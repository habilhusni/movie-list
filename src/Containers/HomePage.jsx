import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import SearchBar from "../Components/SearchBar";
import TableList from "../Components/TableList";
import actions from "../actions";

const HomePage = (props) => (
  <div>
    <SearchBar searchMovies={props.getMovies} />
    <TableList movies={props.movies} getDetails={props.getDetails} />
  </div>
);

HomePage.propTypes = {
  getMovies: PropTypes.func.isRequired,
  getDetails: PropTypes.func.isRequired,
  movies: PropTypes.object,
};

HomePage.defaultProps = {
  movies: {},
};

const mapStateToProps = (state) => ({
  movies: state.movies,
});

const mapDispatchToProps = (dispatch) => ({
  getMovies: (keyword) => dispatch(actions.getMovies(keyword)),
  getDetails: (imdbID) => dispatch(actions.getDetails(imdbID)),
});

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
