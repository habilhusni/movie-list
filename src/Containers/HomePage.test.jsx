import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import store from "../common/mockStore";

import HomePage from "./HomePage";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Snapshot and Functional", () => {
  let component;

  beforeEach(() => {
    component = shallow(<HomePage store={store} />);
  });

  it("Testing HomePage Snapshot", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });
});
