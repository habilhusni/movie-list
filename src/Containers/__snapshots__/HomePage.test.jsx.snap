// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot and Functional Testing HomePage Snapshot 1`] = `
<ContextProvider
  value={null}
>
  <HomePage
    getDetails={[Function]}
    getMovies={[Function]}
    modal={
      Object {
        "isShow": false,
        "source": "",
      }
    }
    movies={
      Object {
        "error": Object {},
        "isLoading": false,
        "payload": Object {},
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    willModalShow={[Function]}
  />
</ContextProvider>
`;
